src/core/mac/sub_mac.cpp

```diff
@@ -315,7 +386,11 @@ void SubMac::BeginTransmit(void)
 
     OT_UNUSED_VARIABLE(error);
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    VerifyOrExit(mState == kStateCsmaBackoff || mState == kStateCslTransmit, OT_NOOP);
+#else
     VerifyOrExit(mState == kStateCsmaBackoff, OT_NOOP);
+#endif
 
     mTransmitFrame.SetCsmaCaEnabled(true);
```

Comment:
This line enables CSMA-CA for CSL frame, too.

Reply:
Do you mean a CSL frame shouldn't have `CsmaCsEnabled` being true? I'm not sure actually. I think it's unified for all tx as it's the final step for a tx, no matter direct, indirect or csl. But again I'm not sure.

Reply:
Per Thread 1.2 Spec section 3.2.6.3.4

> The CSL Transmitter SHALL start transmitting the first symbol of the preamble of the frame to transmit at the time specified in the CSL Phase (that is, CCA must be performed before).

I think in order to make sure the first symbol of preamble to fall at begin of CSL sample, it is not possible to do random backoff. But only CCA should be performed before transmission, that is the frame should arrive at Radio 192us before the sample start point.

I think we can keep the `CsmaCsEnabled` true here, but radio layer need to add some logic to turn off random backoff but only do CCA if cslPhase is set in `TxInfo`.

Reply:
Have updated the implementation. For CSL, we would always set `CsmaCsEnabled` to be `false` in `SubMac`. And in radio, we need to add logic to enable `CCA` to be `true` if the frame's csl period is not 0.